### PR TITLE
fix: limit LSP opening fees for JIT channel invoices

### PR DESCRIPTION
**Pull request by rolznz**
JIT channel invoices were created without a maximum LSP opening fee (`nil` passed to `ReceiveViaJitChannel`), so the fee deducted from an incoming payment was whatever the LSP decided when the payment arrived.

Invoices are now created with an explicit fee limit:

- the highest fee the LSP's advertised LSPS2 opening fee menu allows for the payment size,
- bounded by an absolute ceiling of `max(5000 sats, 10% of the payment)`,
- falling back to the ceiling alone when the fee menu is unavailable or doesn't cover the amount.

ldk-node rejects invoice creation if the LSP quotes a fee above this limit, and enforces the stored limit when the payment is claimed.

The minimum JIT payment size calculation now applies the same ceiling, so the advertised receivable range stays consistent with what invoice creation accepts (e.g. if the LSP's minimum fee exceeds 5000 sats, the minimum payment size becomes the amount at which the fee is 10%). Menus whose fees can never fit the ceiling yield no receivable range.

Includes unit tests for the fee limit computation and the ceiling-aware minimum payment size.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved LSPS2 opening-fee calculations by selecting the applicable advertised fee.
  * Enforced maximum opening-fee limits for JIT invoices and minimum payments.
  * Improved handling of payment-size thresholds, stale fee data, empty fee menus, and invalid fee configurations.
  * Added validation to reject invoice expiry values outside the supported 24-hour range.

* **Tests**
  * Added comprehensive coverage for fee calculation, payment filtering, fee limits, expiry validation, and edge cases.

### CODE REVIEW

File: lnclient/ldk/ldk.go [hunk 1]
```diff
@@ -2662,24 +2676,36 @@ func (ls *LDKService) GetLiquiditySourceLsps2() string {
...
+// getLsps2MaxTotalOpeningFeeMsat returns the maximum opening fee to accept
+// for a JIT channel invoice of the given payment size, derived from the
+// LSP's advertised opening fee menu and an absolute ceiling.
+func (ls *LDKService) getLsps2MaxTotalOpeningFeeMsat(paymentSizeMsat uint64) uint64 {
+	ls.fetchLsps2OpeningFeeParams(lsps2FeeCapCacheTTL)
+
+	ls.lsps2InfoMu.Lock()
+	defer ls.lsps2InfoMu.Unlock()
+
+	return computeLsps2MaxTotalOpeningFeeMsat(paymentSizeMsat, ls.lsps2OpeningFeeParamsMenu)
```
[the remark this answers is not in the document]
Comment (rolznz) [reply lnclient/ldk/ldk.go:2705]: Skipping this one: if the LSP is unreachable, the JIT payment won't go through anyway, so falling back to the absolute ceiling when the cached menu is stale doesn't add protection — it would only loosen the fee cap to the maximum without any LSP-confirmed params behind it. Using the last known menu is the safer behavior here.

File: lnclient/ldk/ldk.go [hunk 2]
```diff
@@ -2708,11 +2734,46 @@ func (ls *LDKService) fetchLsps2OpeningFeeParams() {
...
+		return *menuMaxFeeMsat
+	}
+	return maxAcceptableFeeMsat
+}
+
+// the absolute ceiling on the LSPS2 opening fee for a payment of the given
+// size, independent of the fees the LSP advertises
+func lsps2MaxAcceptableOpeningFeeMsat(paymentSizeMsat uint64) uint64 {
+	return max(lsps2MaxOpeningFeeBaseMsat, paymentSizeMsat/100*lsps2MaxOpeningFeePercent)
+}
```
[the remark this answers is not in the document]
Comment (rolznz) [reply lnclient/ldk/ldk.go:2780]: The truncation is inconsequential here: with a 10% ceiling the worst-case difference is (paymentSizeMsat % 100) * 10 / 100 ≈ 9.9 msat — a hundredth of a sat on a cap that is at minimum 5,000 sats. It also rounds the ceiling down, so the cap only gets marginally stricter, never looser. Dividing first also can't overflow uint64, so keeping the current order.
